src/test/java/teammates/test/cases/browsertests/AdminSearchPageUiTest.java

```diff
@@ -189,22 +217,14 @@ private boolean isSearchTableHeaderCorrect(int tableNum) {
      *            a registered instructor with co-owner privileges or the
      *            privilege to modify instructors
      */
-    private boolean isStudentRowDisplayed(StudentAttributes student, InstructorAttributes instructorToMasquaradeAs) {
-
-        By by = By.xpath("//table[@id = 'search_table']/tbody/tr[@class='studentRow']");
-        List<WebElement> studentRows = browser.driver.findElements(by);
-
-        for (WebElement studentRow : studentRows) {
+    private void assertStudentRowDisplayed(StudentAttributes student, InstructorAttributes instructorToMasquaradeAs,
+                                              CourseAttributes course) {
 
-            boolean isStudentCorrect = isStudentContentCorrect(studentRow, student)
-                                       && isStudentLinkCorrect(studentRow, student, instructorToMasquaradeAs);
-
-            if (isStudentCorrect) {
-                return true;
-            }
-        }
+        By by = By.xpath("//table[@id = 'search_table']/tbody/tr[@id='" + AdminSearchPageData.createId(student) + "']");
+        WebElement studentRow = browser.driver.findElement(by);
```

Comment:
Is it possible to push this kind of code into the relevant `*Page` class itself? i.e. the `*Page` object should provide methods to query its contents. Can be a separate PR though.